text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/placeholder/PlaceholderResolver.java

```diff
@@ -228,20 +151,17 @@
   }
 
   /**
-   * Checks if this placeholder resolver can resolve a placeholder from a key.
+   * Returns the replacement for a given key, if any exist.
    *
-   * @param key the key
-   * @return if a placeholder can be resolved from this key
-   * @since 4.10.0
-   */
-  boolean canResolve(final @NotNull String key);
-
-  /**
-   * Returns a placeholder from a given key, if any exist.
+   * <p>
+   *   This method might be called multiple times during each parse attempt. This is due to the
+   *   fact that it is used in places to check if a tag is a placeholder or not. Therefore, you
+   *   should prefer using fixed or cached replacements instead of dynamic construction.
+   * </p>
```

Comment:
```suggestion
   * <p>This method might be called multiple times during each parse attempt. This is due to the
   *   fact that it is used in places to check if a tag is a placeholder or not. Therefore, you
   *   should prefer using fixed or cached replacements instead of dynamic construction.</p>
```